### ISSUE TITLE
[feature request] Stream video directly to FS

### ISSUE THREAD

**Issue by AdamTheLang** [keybase/client#24344]
There ought to be a solution for taking video and streaming it in as close to real-time as possible to a secure location (keybase's fs). This has some obvious applications with regard to documenting repressive regimes where your phone might be confiscated at any time, but you don't want your video to be easily accessible by authorities via one of the mainstream streaming sites.

Adding a camera interface to the iOS client for this would be relatively easy, but I have no insight into how easy adding the local buffering and real-time uploading would be.

**Comment by AlbinoGeek** [comment]
Keybase [as far as I can tell, based on reaching through the codebase] signs and uploads messages to s3, it doesn't appear to support any kind of streaming currently (whether it be a streamed file transfer, video file, audio, etc.)

**Comment by AdamTheLang** [comment]
That's not a particular problem: chopping your video up into bits and uploading them as each bit is finished is also not especially difficult.

**Comment by AlbinoGeek** [comment]
I'd agree. I was trying to implement something like this before, but other than their `chat api` command, it was feasibly impossible to upload or download a file -- while a file transfer gives you many fields, none of them is the secret required to access the file in the bucket (would love if a dev would explain this process.)

```json
{
  "region":    "us-east-1",
  "endpoint":  "https://s3.amazonaws.com",
  "bucket":    "kb-chat",
  "path":      "-snip-",
  "size":      12345,
  "mimeType":  "audio/x-m4a",
  "encHash":   "-snip-",
  "ptHash":    "-snip-",
  "key":       "-snip-",
  "verifyKey": "-snip-",
  "nonce":     "-snip-"
}
```

Going over the S3 documentation, I could not figure out how I would assemble a signed HMAC256 using only these pieces -- nor could I figure out if one of these keys were already that.

**Comment by AdamTheLang** [comment]
Wait, I'm confused. What about that keybase filesystem thing? That's all files.

**Comment by AlbinoGeek** [comment]
Right, this was a copy of the API message given when an attachment hit chat, which then made me realize that all chat messages end up in that S3 bucket -- thereby thinking that all of the files in KBFS also went there. That, and, we [my company] were going to implement video chat using team channels and chunked uploads, hence going down this rabbit hole.
